### PR TITLE
Add functionality for multiple visualisations

### PR DESCRIPTION
**Pull request by arsalanc-v2**
Changes:

- More charts
- Ability to switch between them

### CODE REVIEW

**Comment by jjlee050** [comment]
screenshot 2018-11-06 at 18 38 51
I
Wow. It work along the status bar footer however the thing still look covered plus the white space below. In addition, the next & previous button does not do anything.

File: src/main/java/seedu/clinicio/ui/ResultDisplay.java [hunk 1]
```diff
@@ -11,6 +11,7 @@
 import javafx.fxml.FXML;
 import javafx.scene.control.TextArea;
 import javafx.scene.layout.Region;
+import javafx.scene.layout.StackPane;
```
Comment (jjlee050) [inline src/main/java/seedu/clinicio/ui/ResultDisplay.java]: Did you add this import without using it?

File: src/main/java/seedu/clinicio/ui/analytics/Plot.java [hunk 2]
```diff
@@ -68,16 +54,22 @@ public static void plotChart(VisualizationData data, Pane chartPane) {
         case VERTICAL_BAR:
             plotVBar((VisualizationData<String>) data, chartPane);
             break;
+
```
Comment (jjlee050) [inline src/main/java/seedu/clinicio/ui/analytics/Plot.java]: The new line can remove

Comment (jjlee050) [reply src/main/java/seedu/clinicio/ui/analytics/Plot.java]: Same for the bottom few

File: src/main/java/seedu/clinicio/ui/analytics/Plot.java [hunk 3]
```diff
@@ -87,22 +79,22 @@ public static void plotChart(VisualizationData data, Pane chartPane) {
      * Plots a stacked bar chart.
      * Assumes a minimum of two groups of data.
      */
-    public static void plotStackedBar(VisualizationData<String> barsData, Pane chartPane) {
+    public static void plotStackedBar(VisualizationData<String> data, Pane chartPane) {
```
Comment (jjlee050) [inline src/main/java/seedu/clinicio/ui/analytics/Plot.java]: This method is very long. (Req: <20 lines). Can work on modularising part in here.

File: src/main/java/seedu/clinicio/ui/analytics/Plot.java [hunk 4]
```diff
@@ -117,77 +109,103 @@ public static void plotStackedBar(VisualizationData<String> barsData, Pane chart
     /**
      * Plots a vertical bar chart for categorical data.
      */
-    public static void plotVBar(VisualizationData<String> barsData, Pane chartPane) {
-        CategoryAxis xAxis = new CategoryAxis();
-        xAxis.setLabel(barsData.getXTitle());
+    public static void plotVBar(VisualizationData<String> data, Pane chartPane) {
```
Comment (jjlee050) [inline src/main/java/seedu/clinicio/ui/analytics/Plot.java]: Same as above (Req: <20 lines)

Comment (jjlee050) [reply src/main/java/seedu/clinicio/ui/analytics/Plot.java]: In addition, there is some part here plotStackedBar and plotHBar that can be extracted out to modularise.

File: src/main/java/seedu/clinicio/ui/analytics/Plot.java [hunk 5]
```diff
@@ -117,77 +109,103 @@ public static void plotStackedBar(VisualizationData<String> barsData, Pane chart
...
         chartPane.getChildren().add(barChart);
     }
+
+    /**
+     * Plots a line chart for continuous data.
+     * May have String or Integer labels on x axis.
+     * For String labels, data points must consist of String, Integer pairs. The String components of all the pairs that
+     * match the supplied labels will be shown as the  x axis tick labels while those that don't match will be removed.
+     */
+    public static void plotLine(VisualizationData data, Pane chartPane) {
```
Comment (jjlee050) [inline src/main/java/seedu/clinicio/ui/analytics/Plot.java]: Same (Req: <20 lines)

File: src/main/java/seedu/clinicio/ui/MainWindow.java [hunk 6]
```diff
@@ -144,24 +132,13 @@ private void setAccelerator(MenuItem menuItem, KeyCombination keyCombination) {
      * Fills up all the placeholders of this window.
      */
     void fillInnerParts() {
-
-        appointmentListPanel = new AppointmentListPanel(logic.getFilteredAppointmentList());
```
Comment (gingivitiss) [inline src/main/java/seedu/clinicio/ui/MainWindow.java]: why was there this removal?

Comment (arsalanc-v2) [reply src/main/java/seedu/clinicio/ui/MainWindow.java]: Was trying another mechanism to switch the panels then thought you'd add it back in once you're done with the switching logic.

Comment (arsalanc-v2) [reply src/main/java/seedu/clinicio/ui/MainWindow.java]: Added it back along with toggling events

**Pull request merged**